src/main/java/se/lth/cs/connect/modules/AccountSystem.java

```diff
@@ -182,15 +184,65 @@ public static synchronized boolean createAccount(String email,
      * Try and remove the user from the database. The user and all relations
      * to entries will be deleted. Methods throws database exception on error.
      */
-    public static boolean deleteAccount(String email) {
+    public static boolean deleteAccount(String email, IDBAccess db) {
         JcNode node = new JcNode("user");
-        JcQueryResult res = Database.query(Database.access(), new IClause[]{
+        JcNode coll = new JcNode("coll");
+        JcNode inviters = new JcNode("inviter");
+        JcRelation rel = new JcRelation("relation");
+        
+        //find all collection user is owner of
+        //from those collection go back to all owners <----------------------|
+        //check if any of the owners have outgoing invites <-----------------|
+        //delete invites													 |
+        //delete collection and all relations								 |
+        //-------------------------------------------------------------------|
+        //these might become obsolete later if only one owner is allowed-----|
+       Database.query(db, new IClause[]{
+			MATCH.node().label("user").property("email").value(email)
+            	.relation().type("OWNER")
```

Comment:
Ok this is good. I feel like an event system is more and more beneficial for these kinds of clean ups.